src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,190 @@
+
+
+        try:
+            self._TRANSLATION_XYZ_TOLERANCE = rospy.get_param('~translation_xyz_tolerance')
+            self._TRANSFORM_TIMEOUT = rospy.get_param('~transform_timeout')
+            self._MIN_MANEUVER_HEIGHT = rospy.get_param('~min_maneuver_height')
+            self._MAX_HORIZ_SPEED = rospy.get_param('~max_translation_speed')
+            self._MAX_TASK_DIST = rospy.get_param('~max_roomba_dist')
+            self._K_X = rospy.get_param('~k_term_tracking_x')
+            self._K_Y = rospy.get_param('~k_term_tracking_y')
+            self._z_position = rospy.get_param('~track_roomba_height')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for xyztranslation task')
+            raise
+
+        # Check that we aren't being requested to go below the minimum maneuver height
+        # Error straight out if that's the case. If we are currently below the minimum height
+        # It will be caught and handled on the next update
+        if self._z_position < self._MIN_MANEUVER_HEIGHT :
+            raise ValueError('Requested z height was below the minimum maneuver height')
+
+        # fc status?
+        self._path_holder = TranslateStopPlanner(self._x_position,
+                                                    self._y_position,
+                                                    self._z_position)
+
+        self._state = GoToRoombaState.init
+
+
+    def get_desired_command(self):
+        with self._lock:
+            if self._canceled:
+                return (TaskCanceled(),)
+            if self._state == GoToRoombaState.init:
+                self._state = GoToRoombaState.translate
+
+            if self._state == GoToRoombaState.translate:
+                try:
+                    transStamped = self.topic_buffer.get_tf_buffer().lookup_transform(
+                            'map',
+                            'quad',
+                            rospy.Time(0),
+                            rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('GoToRoombaTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg = 'Exception when looking up transform during go_to_roomba'),)
+
+                if(transStamped.transform.translation.z > self._MIN_MANEUVER_HEIGHT):
+                    """ quarantine"""
+                    # check if the roomba is in range
+                    if not self._check_roomba_in_sight():
+                        return (TaskAborted(msg='The provided roomba is not in sight of quad'),)
+                    elif self._on_ground():
+                        return (TaskDone(),)
+
+                    try:
+                        roomba_transform = self.topic_buffer.get_tf_buffer().lookup_transform(
```

Comment:
Same comment as below. Why do we need to lookup any transforms? Why can’t we just use the current roomba position in ‘self._roomba_odometry‘? Is it to guarantee the appropriate frame?